app/src/main/java/com/jerboa/nav/NavigateImplementations.kt

```diff
@@ -0,0 +1,17 @@
+package com.jerboa.nav
+
+import androidx.lifecycle.ViewModel
+
+class NavigateWithNoArgsAndDependencies(
+    val navigate: () -> Unit,
+)
+
+class NavigateWithNoArgs<D : ViewModel>(
+    private val container: DependencyContainer<D>,
+    private val navigateToDestination: () -> Unit,
+) {
+    fun navigate(dependencies: D) {
+        container.dependencies = dependencies
```

Comment:
Honestly, did not think about process death when implementing this. 

The intention:

Take a look at the PostEditViewModel in the main branch:
https://github.com/dessalines/jerboa/blob/30aa860bf40e2e7b522e9abb590ae41223085cd6/app/src/main/java/com/jerboa/ui/components/post/edit/PostEditViewModel.kt#L53-L59

Since the post can be edited from the home (or feed) activity, post activity, community activity & a person's profile activity, all these view models have to be passed to the PostEditViewModel to update the data. Usually after editing a post, you go back to any one of the these activities and it's good UX to show the updated data (for confirmation etc). Also the update logic should ideally not be in the post edit view model.

My intention was to invert this:
With this PR one can pass a callback for post edit and any activity can update itself accordingly.

In home/feed activity:
https://github.com/nahwneeth/jerboa/blob/bcefa3a67e8d81ac4398b70b2afa393dafa04028/app/src/main/java/com/jerboa/ui/components/home/FeedActivity.kt#L320-L327
```kotlin
onEditPostClick = { postView ->
    navController.toPostEdit.navigate(
        PostEditDependencies(
            postView = postView,
            onPostEdit = homeViewModel::updatePost,
        ),
    )
},
```

In post activity:
https://github.com/nahwneeth/jerboa/blob/bcefa3a67e8d81ac4398b70b2afa393dafa04028/app/src/main/java/com/jerboa/ui/components/post/PostActivity.kt#L254-L261
```
onEditPostClick = { pv ->
    navController.toPostEdit.navigate(
        PostEditDependencies(
            postView = pv,
            onPostEdit = postViewModel::updatePost,
        ),
    )
}
```

Person profile activity:
https://github.com/nahwneeth/jerboa/blob/bcefa3a67e8d81ac4398b70b2afa393dafa04028/app/src/main/java/com/jerboa/ui/components/person/PersonProfileActivity.kt#L381-L388
```
onEditPostClick = { pv ->
    navController.toPostEdit.navigate(
        PostEditDependencies(
            postView = pv,
            onPostEdit = personProfileViewModel::updatePost,
        ),
    )
},
```

I hope u get the idea. I tried a few things and I felt this was the best way to pass dependencies/callbacks. Of course, there might be a better way to do this which I'm not aware of. Open to suggestions and ideas.

The second intention is to scope the view models to the routes:
Right now these view models are declared globally and they are reused. This has resulted in stale data and the workaround currently being used to reload the view models when navigating back which is leading to other issues like the data getting reset, scroll positions not being retained and filters getting reset. These global view models also, if i'm not wrong, will not survive process death.